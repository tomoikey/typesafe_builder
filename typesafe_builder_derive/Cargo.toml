[package]
name = "typesafe_builder_derive"
version = "1.3.0"
edition = "2024"
authors = ["tomoikey"]
readme = "README.md"
repository = "https://github.com/tomoikey/typesafe_builder"
categories = ["accessibility", "development-tools", "rust-patterns"]
license = "MIT"
description = "A procedural macro to generate type-safe builder patterns for Rust structs"

[lib]
proc-macro = true

[dependencies]
darling = "0.20.11"
proc-macro2 = "1.0.95"
syn = { version = "2.0.101", features = ["derive", "parsing", "proc-macro"] }
quote = "1.0.40"

[dev-dependencies]
trybuild = "1.0.105"
