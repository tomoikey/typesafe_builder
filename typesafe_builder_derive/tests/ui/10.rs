use typesafe_builder_derive::Builder;

struct _TypesafeBuilderEmpty;
struct _TypesafeBuilderFilled;

fn main() {
    #[derive(Builder)]
    struct User {
        #[builder(optional, required)]
        name: String,
    }
} 