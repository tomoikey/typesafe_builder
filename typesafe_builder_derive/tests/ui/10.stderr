error: Invalid requirement: Multiple builder attributes specified. Only one of required, optional, required_if, optional_if, default is allowed
 --> tests/ui/10.rs:7:14
  |
7 |     #[derive(Builder)]
  |              ^^^^^^^
  |
  = note: this error originates in the derive macro `Builder` (in Nightly builds, run with -Z macro-backtrace for more info)
